content/solids/components/net.ts

```diff
@@ -0,0 +1,343 @@
+
+    const bone = new THREE.Bone();
+    bone.name = `${currentFaceIndex}-${faceIndex}`;
+    const parentPos = new THREE.Vector3();
+    previousBone.getWorldPosition(parentPos);
+    const parentPosPoint = new Point(parentPos.x, parentPos.y);
+    const diff = midpoint.subtract(parentPosPoint);
+    bone.position.set(diff.x, diff.y, 0);
+    previousBone.add(bone);
+
+    const verts: Point[] =
+      nextFace.points.filter(point =>
+        !point.equals(hingeEdge.p1) &&
+        !point.equals(hingeEdge.p2)
+      );
+
+    const hinge = hingeForFaces(faceIndex, currentFaceIndex, hinges);
+    const [_a, _b, maxAngleDeg] = hinge!;
+    const maxAngle = angleFromDegrees(maxAngleDeg).rad;
+
+    const boneHinge = {
+      bone,
+      verts,
+      axis,
+      maxAngle
+    };
+
+    const children = acc.concat(
+        boneHingesForFace(
+            faceIndex,
+            currentFaceIndex,
+            bone,
+            faces,
+            hinges
+        )
+    );
+    return [boneHinge, ...children];
+  },
+  []
+  );
+
+  return boneHinges;
+}
+/**
+  * Assumes that `a` and `b` are adjacent
+  */
+function getCommonEdge(a: Polygon, b: Polygon) {
+  return a.edges.filter(aEdge => b.edges.some(bEdge => aEdge.equals(bEdge)))[0];
+}
+
+function getBoneIndexFromPoint(point: Point, hingeBones: HingeBone[]) {
+  return hingeBones.findIndex(
+      hingeBone => hingeBone.verts.some(
+          vert => vert.equals(point)
+      )
+  );
+}
+
+function hingesForFace(faceIndex: number, hinges: Hinge[]) {
+  return hinges.filter(hinge =>
+    hinge[0] == faceIndex || hinge[1] == faceIndex
+  );
+}
+
+function hingeForFaces(a: number, b: number, hinges: Hinge[]) {
+  return hinges.find(hinge =>
+    (hinge[0] == a && hinge[1] == b) ||
+    (hinge[1] == a && hinge[0] == b)
+  );
+}
+
+function otherFaceIndex(currentFaceIndex: number, hinge: Hinge) {
+  if (currentFaceIndex == hinge[0]) return hinge[1];
+  else return hinge[0];
+}
+
+function mergePolys(polys: Polygon[]) {
+  const pointSet = new PointSet();
+  for (const poly of polys) {
+    pointSet.add(poly.points);
+  }
+  return new Polygon(...pointSet.points);
+}
+
+function triangulateFace(face: Polygon): Array<[Point, Point, Point]> {
+  const verts = face.points;
+  return verts.slice(1, verts.length - 1).map((vert, index) => {
+    const a = new Point(verts[0].x, verts[0].y);
+    const b = new Point(vert.x, vert.y);
+    const c = new Point(verts[index + 2].x, verts[index + 2].y);
+    return [a, b, c];
+  });
+}
+
+class PointSet {
+  private _points: Point[] = [];
+
+  constructor(points?: Point[]) {
+    if (points != undefined) this.add(points);
+  }
+
+  add(points: Point[]) {
+    for (const point of points) {
+      this.push(point);
+    }
+  }
+
+  push(point: Point) {
+    const isNew = this.points.every(p => !p.equals(point));
+    if (isNew) this.points.push(point);
+  }
+
+  get points() {
+    return this._points;
+  }
+
+}
+
+function angleFromDegrees(val: number): Angle {
```

Comment:
Do you want to import these two functions from euclid.js?

Reply:
The euclid.js version (0.6.8) with these is not currently on npm as far as I can tell.

Reply:
Oops, sorry – I forgot to click publish. Should be fixed now, but you can also change this in a later PR.